src/main/java/com/dnd/runus/infrastructure/persistence/jooq/scale/JooqScaleRepository.java

```diff
@@ -25,6 +35,34 @@ public ScaleSummary getSummary() {
                 .fetchOne(new ScaleSummaryMapper());
     }
 
+    public List<Long> findAchievableScaleIds(long memberId) {
+
+        CommonTableExpression<Record1<Integer>> totalDistance = name("total_distance")
+                .fields("total_distance_meter")
+                .as(select(sum(RUNNING_RECORD.DISTANCE_METER).cast(int.class))
+                        .from(RUNNING_RECORD)
+                        .where(RUNNING_RECORD.MEMBER_ID.eq(memberId)));
+
+        CommonTableExpression<Record2<Long, Integer>> cumulativeScale = name("cumulative_scale")
+                .fields("id", "cumulative_sum")
+                .as(select(
+                                SCALE.ID,
+                                sum(SCALE.SIZE_METER).over().orderBy(SCALE.ID).cast(int.class))
+                        .from(SCALE)
+                        .where(SCALE.ID.notIn(select(SCALE_ACHIEVEMENT.SCALE_ID)
+                                .from(SCALE_ACHIEVEMENT)
+                                .where(SCALE_ACHIEVEMENT.MEMBER_ID.eq(memberId)))));
+
+        return dsl.with(totalDistance)
+                .with(cumulativeScale)
+                .select(cumulativeScale.field("id", Long.class))
+                .from(cumulativeScale)
+                .join(totalDistance)
+                .on(Objects.requireNonNull(cumulativeScale.field("cumulative_sum", Integer.class))
```

Comment:
coalesce를 사용하면 running_record가 없을때 NPE가 발생하지 않아요

```suggestion
                .on(coalesce(cumulativeScale.field("cumulative_sum", Integer.class), 0)
```